File: 0300. Longest Increasing Subsequence/dp2.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& v) {
        int len = v.size(), ret = 0;
        vector<int> dp = vector<int>(len + 1, 1);
        for(int i = 1; i < len + 1; ++i){
            for(int j = 1; j < i; ++j){
                if(v[i - 1] > v[j - 1]){
                    dp[i] = max(dp[j] + 1, dp[i]);
                }
            }
            ret = max(ret, dp[i]);
        }
        return ret;
    }
};

File: offer/JZ42 连续子数组的最大和/dp2.java
import java.util.Arrays;
public class Solution {
    public int FindGreatestSumOfSubArray(int[] v) {
        if(v.length == 0)
            return 0;
        int len = v.length, ret = 0, tmp = 0;
        for(int i = 0; i < len; ++i){
            tmp = tmp + v[i];
            if(tmp < 0){
                tmp = 0;
            }
            ret = Math.max(ret, tmp);
        }
        return ret != 0? ret : Arrays.stream(v).max().getAsInt();
    }
}